libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -81,6 +81,25 @@ func New(cfg *common.Config) (processors.Processor, error) {
 
 // Run enriches the given event with the host meta data
 func (p *addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
+	// check replace_host_fields field
+	if !p.config.ReplaceHostFields {
+		// If host fields exist(besides host.name added by libbeat) in event, skip add_host_metadata.
+		hostFields, err := event.Fields.GetValue("host")
+		if err == nil && hostFields != nil && len(hostFields.(common.MapStr)) >= 1 {
+			if hasName, _ := hostFields.(common.MapStr).HasKey("name"); hasName {
+				// other host fields exist on top of host.name, skip add_host_metadata.
+				if len(hostFields.(common.MapStr)) > 1 {
+					return event, nil
+				}
+			} else {
+				// host.name does not exist but other host fields exist, skip add_host_metadata.
+				if len(hostFields.(common.MapStr)) > 0 {
+					return event, nil
+				}
+			}
+		}
```

Comment:
This check is starting to grow, consider moving this logic to a different method.

Reply:
Good point, will do!